frontend/src/tests/posting-tests.js

```diff
@@ -1,39 +1,356 @@
+            ],
+        },
+    };
+    const postingPos = {
+        hours: 20,
+        num_positions: 1,
+    };
+
     beforeAll(async () => {
         await databaseSeeder.seed(api);
+        session = databaseSeeder.seededData.session;
+        position = databaseSeeder.seededData.position;
+        session2 = await addSession(api);
+    });
+
+    it("Create a posting for a session", async () => {
+        resp = await apiPOST("/admin/postings", {
+            ...posting,
+            session_id: session.id,
+        });
+        expect(resp).toHaveStatus("success");
+        checkPropTypes(postingPropTypes, resp.payload);
+        expect(resp.payload.id).not.toBeNull();
+        Object.assign(posting, resp.payload);
+
+        const newPosting = {
+            name: "CSC200F TA",
+            intro_text: "Testing posting for CSC200F",
+            open_date: "2021/04/01",
+            close_date: "2021/05/01",
+            availability: "open",
+        };
+
+        resp = await apiPOST(
+            `/admin/sessions/${session.id}/postings`,
+            newPosting
+        );
+        expect(resp).toHaveStatus("success");
+        checkPropTypes(postingPropTypes, resp.payload);
+        expect(resp.payload.id).not.toBeNull();
+    });
+
+    // MUST run "Create a posting for a session" first to load data into ‘posting’
+    it("Modify a posting", async () => {
+        const updatedPostingData = {
+            id: posting.id,
+            open_date: "2021-06-01T00:00:00.000Z",
+            close_date: "2021-08-01T00:00:00.000Z",
+            availability: "open",
+        };
+        resp = await apiPOST("/admin/postings", updatedPostingData);
+        expect(resp).toHaveStatus("success");
+        expect(resp.payload).toMatchObject(updatedPostingData);
+
+        const updatedPostingData2 = {
+            id: posting.id,
+            open_date: "2021-07-01T00:00:00.000Z",
+            close_date: "2021-09-01T00:00:00.000Z",
+            availability: "auto",
+        };
+        resp = await apiPOST(
+            `/admin/sessions/${session.id}/postings`,
+            updatedPostingData2
+        );
+        expect(resp).toHaveStatus("success");
+        expect(resp.payload).toMatchObject(updatedPostingData2);
+        Object.assign(posting, resp.payload);
+    });
+
+    it("Cannot set the `status` of a posting to an invalid value", async () => {
+        resp = await apiPOST("/admin/postings", {
+            ...posting,
+            availability: "test",
+        });
+        expect(resp).toHaveStatus("error");
+
+        resp = await apiPOST("/admin/postings", {
+            ...posting,
+            availability: 3,
+        });
+        expect(resp).toHaveStatus("error");
+    });
+
+    it("Fetch all postings for a session", async () => {
+        resp = await apiGET(`/admin/sessions/${session.id}/postings`);
+        expect(resp).toHaveStatus("success");
+        checkPropTypes(PropTypes.arrayOf(postingPropTypes), resp.payload);
```

Comment:
You know what postings should be here. You should check for them explicitly. Also, does any session have more than one posting? You should make sure there's a session with more than one posting before checking.

Reply:
Yes. The pre-seeded `session` should contain two postings at this point.